File: devicems/src/main/java/com/example/devicems/device/DeviceService.java
package com.example.devicems.device;

import java.util.List;

public interface DeviceService {

    List<Device> findAll();
    void createDevice(Device device);
    Device getDeviceById(Long id);
    boolean updateDevice(Long id, Device device);
    boolean deleteDeviceById(Long id);

    List<Device> findDevicesByUserId(Long userId);
}

File: react-demo/src/services/DeviceService.js
import axios from 'axios';

const DEVICE_API_BASE_URL = "http://device-service-1.localhost/device";
class DeviceService {

    getDevices(){
        return axios.get(DEVICE_API_BASE_URL);
    }
    createDevice(device) {
        return axios.post(DEVICE_API_BASE_URL, device);
    }

    getDeviceById(deviceId) {
        return axios.get(DEVICE_API_BASE_URL + '/' + deviceId);
    }

    updateDevice(device, deviceId){
        return axios.put(DEVICE_API_BASE_URL + '/' + deviceId, device);
    }

    deleteDevice(deviceId)   {
        return axios.delete(DEVICE_API_BASE_URL + '/' + deviceId);
    }

    getDevicesByUserId(userId) {
        return axios.get(DEVICE_API_BASE_URL  + `/user/${userId}`);
    }

}

export default new DeviceService()